includes/enrolment/class-sensei-course-enrolment.php

```diff
@@ -93,6 +93,10 @@ public function get_course_id() {
 	 * @return bool
 	 */
 	public function is_enrolled( $user_id, $check_cache = true ) {
+		if ( empty( $user_id ) ) {
```

Comment:
Curious, why is `empty()` better than usual `!` here?

Reply:
I usually opt for `empty()`, although in this case it is the same as `!`. What is the benefit of `!`?

Reply:
https://www.php.net/manual/en/function.empty.php says:

> No warning is generated if the variable does not exist. That means empty() is essentially the concise equivalent to !isset($var) || $var == false.

* `!` alone is less cognitive load
* Clarity that this isn't some special thing that `empty()` extra oompf is needed
* Is there a codebase-wide convention for these yet? Either way, I follow a lot of code and almost never see `empty()` for non-array situations, hence the curiosity.

Reply:
> No warning is generated if the variable does not exist. That means empty() is essentially the concise equivalent to !isset($var) || $var == false.

Yup, that is what I meant by `although in this case it is the same as !`. I could change it, but it doesn't seem necessary. Even though here the var will always exist, `empty()` seems like a safer convention, which is why I use it.